src/db.rs

```diff
@@ -1,297 +1,533 @@
+//! the review chats chat_id stored in a dedicated field.
+//!
+//! See migrations folder for further details.
+
+use std::path::PathBuf;
+
 use anyhow::Context;
-use deltachat::{chat::ChatId, contact::ContactId};
-use serde::{Deserialize, Serialize};
-use serde_json::json;
-use surrealdb::{
-    engine::local::{Db, File},
-    sql::Thing,
-    Surreal,
-};
-use ts_rs::TS;
+use deltachat::{chat::ChatId, contact::ContactId, message::MsgId};
+use sqlx::{migrate::Migrator, Connection, FromRow, Row, SqliteConnection};
 
 use crate::{
     bot::BotConfig,
     request_handlers::{review::ReviewChat, submit::SubmitChat, AppInfo, ChatType},
 };
 
-#[derive(Serialize, Deserialize)]
-struct DBChatType {
-    chat_type: ChatType,
-}
+pub static MIGRATOR: Migrator = sqlx::migrate!();
 
-#[derive(Serialize, Deserialize)]
-struct DBContactId {
-    contact_id: ContactId,
-}
-
-#[derive(Serialize, Deserialize)]
-struct SerialReps {
-    serial: usize,
-}
-
-#[derive(Deserialize, Serialize, Clone, Debug)]
-pub struct AppInfoId {
-    #[serde(flatten)]
-    pub app_info: AppInfo,
-    pub id: Thing,
-}
-
-#[derive(Deserialize, Serialize, Clone, Debug, TS)]
-#[ts(export)]
-#[ts(export_to = "frontend/src/bindings/")]
-pub struct FrontendAppInfo {
+#[derive(FromRow)]
+pub struct DBAppInfo {
+    pub id: RecordId,
     pub name: String,                    // manifest
     pub author_name: String,             // bot
     pub author_email: String,            // bot
     pub source_code_url: Option<String>, // manifest
     pub image: Option<String>,           // webxdc
     pub description: Option<String>,     // submit
+    pub xdc_blob_dir: Option<String>,    // bot
     pub version: Option<String>,         // manifest
-    pub id: String,
+    pub originator: RecordId,            // bot
+    pub active: bool,                    // bot
 }
 
-impl From<AppInfoId> for FrontendAppInfo {
-    fn from(app_info: AppInfoId) -> Self {
+impl From<DBAppInfo> for AppInfo {
+    fn from(db_app: DBAppInfo) -> Self {
         Self {
-            name: app_info.app_info.name,
-            author_name: app_info.app_info.author_name,
-            author_email: app_info.app_info.author_email,
-            source_code_url: app_info.app_info.source_code_url,
-            image: app_info.app_info.image,
-            description: app_info.app_info.description,
-            version: app_info.app_info.version,
-            id: app_info.id.id.to_raw(),
+            id: db_app.id,
+            name: db_app.name,
+            author_name: db_app.author_name,
+            author_email: db_app.author_email,
+            source_code_url: db_app.source_code_url,
+            image: db_app.image,
+            description: db_app.description,
+            xdc_blob_dir: db_app.xdc_blob_dir.map(PathBuf::from),
+            version: db_app.version,
+            originator: db_app.originator,
+            active: db_app.active,
```

Comment:
is it possible to use sth like `Self {...db_app, xdc_blob_dir: db_app.xdc_blob_dir.map(PathBuf::from)}` here?

Reply:
mismatched types expected `AppInfo`, found `DBAppInfo`